jaeger-micrometer/README.md

```diff
@@ -0,0 +1,153 @@
+# Micrometer integration for Jaeger internal metrics
+
+This module brings a Micrometer integration to the internal Jaeger metrics.
+
+## Configuring
+
+To configure the Micrometer integration, a custom Jaeger Tracer instance has to be used:
+
+```java
+MicrometerMetricsFactory metricsReporter = new MicrometerMetricsFactory();
+Configuration configuration = new Configuration("myServiceName");
+Tracer tracer = configuration
+    .getTracerBuilder()
+    .withMetrics(new com.uber.jaeger.metrics.Metrics(metricsReporter))
+    .build();
+```
+
+Ideally, this Tracer would be build during the runtime bootstrap of your application
+and made available to other components via a `GlobalTracer`:
+
+```java
+GlobalTracer.register(tracer);
+```
+
+## Example
+
+The following is a complete Vert.x application demonstrating the Micrometer integration.
+Note that it makes use of the Prometheus Registry as the concrete backend for Micrometer,
+available under the artifact coordinates `io.micrometer:micrometer-registry-prometheus`.
+
+```java
+public class MainVerticle extends AbstractVerticle {
+  private static final Logger logger = LoggerFactory.getLogger(MainVerticle.class);
+
+  public static void main(String[] args) {
+      System.setProperty("JAEGER_SERVICE_NAME", "jaeger-client-java-tester");
+      System.setProperty("JAEGER_REPORTER_LOG_SPANS", "true");
+      System.setProperty("JAEGER_SAMPLER_TYPE", "const");
+      System.setProperty("JAEGER_SAMPLER_PARAM", "1");
+      Vertx vertx = Vertx.vertx();
+      vertx.deployVerticle(new MainVerticle());
+  }
+
+  @Override
+  public void start() {
+    MicrometerMetricsFactory metricsReporter = new MicrometerMetricsFactory();
+    Configuration configuration = Configuration.fromEnv();
```

Comment:
I know @pavolloffay doesn't like mixing `fromEnv()` with `with*` methods, so, I have an alternative to this, but I think a bit more verbose (so, I'd like to avoid, if I could):

```java
    Configuration configuration = new Configuration("jaeger-client-java-tester");
    Tracer tracer = configuration
        .withReporter(
            new Configuration.ReporterConfiguration()
                .withLogSpans(true)
        )
        .withSampler(
            new Configuration.SamplerConfiguration()
                .withType(ConstSampler.TYPE)
                .withParam(1)
        )
.getTracerBuilder()
```

(runnable code: https://github.com/jpkrohling/jaeger-client-java-tester/blob/master/src/main/java/io/vertx/starter/MainVerticle.java#L29-L39)